feat: file explorer adheres to ignores

kas2020-commits opened the pull request:
Currently, the file explorer populates the picker table using `std::fs::read_dir`, with no way to add any filters. This PR refactors the logic out of `file_picker` and reuses it to populate data for `file_explorer`.

Re: if it's proper to reuse the same config options for both the picker and the explorer, _in my humble opinion using the explorer_, I interpret the explorer's main use case as "an interactive version of the picker" in which case you'd always want them to use the same data source. But if new options are required for approval, I can add that.

Also I'm not a rust aficionado and would appreciate someone double-checking my usage of `&`, esp. in the function call sites pls & thanks :) 

fixes #12764

kas2020-commits commented:
The motivation here is that some directories which have code in them also have a lot of hidden files/directories. Being able to apply the same custom filtering for the file explorer helps with the usability a lot.

RoloEdits commented:
I actually prefer to have the split behavior. The file finder hiding hidden things when search is wanted. The explorer is when I actually want to explore the directories without having to switch contexts. Specially when I would want to open something like `.cargo` to access the config.toml in the current helix session.

kas2020-commits commented:
Thanks for the feedback, I've added separate, independent options now for the file explorer and set the defaults to not ignore anything (i.e. same behavior as right now). I also updated the docs accordingly.

Please let me know if it's good to go now :)

kas2020-commits commented:
Friendly ping on getting this reviewed? I've been carrying this patch for almost a month with daily use on my local and it works as hoped.

the-mikedavis reviewed (approved):
While I don't like adding many config options, the file-explorer and file-picker have different use-cases and being able to tune the ignore options independently seems pretty valuable to me, as well as defaulting to showing more rather than less 👍
